.github/workflows/knip.yml

```diff
@@ -1,7 +1,8 @@
 name: Knip
 
 on:
-  pull_request:
+  pull_request_target:
```

Comment:
@gitcoindev 

`pull_request_target` uses context of the target repo, i.e.:
- has access to organization secrets
- checks out the `development` branch [here](https://github.com/ubiquity/keygen.ubq.fi/blob/f8f6dd819f13f80f0342dc91291d82bd9db59fd9/.github/workflows/knip.yml#L14) instead of PR branch

And if you checkout [here](https://github.com/ubiquity/keygen.ubq.fi/blob/f8f6dd819f13f80f0342dc91291d82bd9db59fd9/.github/workflows/knip.yml#L14) the code from PR (with `pull_request_target`) then on `yarn install` [here](https://github.com/ubiquity/keygen.ubq.fi/blob/f8f6dd819f13f80f0342dc91291d82bd9db59fd9/.github/workflows/knip.yml#L22) any fork will be able to get github token with repo write access

Overall `pull_request_target` most of the time opens a security hole, you may check the long read [here](https://github.com/ubiquity/cloudflare-deploy-action/issues/1#issuecomment-1999654318)

Do we really need to use `pull_request_target`? There should be a way to run knip workflow on `pull_request`

Reply:
@rndquu you are right, let me read the thread you mentioned, there must be a way to run with `pull_request`.